clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1227,6 +1227,7 @@ public void close(Duration timeout) {
 
     private void close(Duration timeout, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
+        boolean wasInterrupted = Thread.interrupted();
```

Comment:
I don't think this is correct. With this code, if the thread was interrupted BEFORE the close method was called, `Thread.interrupted` will return true and then we will take different action. I think what we need is that if we were interrupted WHILE the close method was running, then we need to leave the group politely and then throw the `InterruptException`.

Reply:
If the thread was interrupted before close, AddAndGet methods will throw an interrupted exception, so we still need to call Thread.interrupted() on the very top.

If the thread is interrupted during closing, we could catch that exception and leave group as you said.

Reply:
I think we need to handle the case where the thread is interrupted both _before_ and _during_ the call to `close()`. But you're right, looking back at the Jira, it clearly states the _during_ case, which I somehow missed.

Reply:
hey, sorry if my "while closing" on the jira is misleading, but I think the stress test that revealed this was the "before" case: test that had a "finally" clause, where it closed the consumer. Interrupting the thread would lead to that finally, and at that point it seemed that the classic consumer would do a clean leave but the new consumer didn't.

Reply:
@lianetm—what's your take: should I push further on this and implement the _during_ `close()` case too, or handle that in a separate Jira/PR? Not to be too focused on the the short term, but we want to unblock our stress tests.

Reply:
I would suggest we deal with the before case, to ensure an attempt to close sends the leave even if happens after the thread was interrupted. That's the case we got, and it actually seems like a sensible uses case that many apps would attempt: close a consumer in a finally/catch (our test does it in the catch actually).

Reply:
I think we need a solution to interrupt DURING close, but fixing the situation in hand first where the stress tests are blocked makes a lot of sense. The during case can certainly be a follow-on PR.

Reply:
The case we have is
```
        executor.shutdownNow();
       // wait
        consumer.close();
```

in executor.shutdownNow, Thread.interrupt() is invoked.